pkg/skaffold/build/local/docker.go

```diff
@@ -45,7 +45,7 @@ func (b *Builder) buildDocker(ctx context.Context, out io.Writer, workspace stri
 		imageID, err = b.localDocker.Build(ctx, out, workspace, a, tag)
 	}
 
-	if b.pushImages {
+	if err == nil && b.pushImages {
```

Comment:
Instead of checking `if err==nil` , let's instead check

```golang
if err != nil {
   return "", err
}
```
before pushing, and then at the end of the function we can

```golang
return imageID, nil
```

Reply:
Done.